dist/common/scripts/scylla_util.py

```diff
@@ -105,6 +109,141 @@ def curl(url, byte=False):
             if (retries >= max_retries):
                 raise
 
+class gcp_instance:
+    """Describe several aspects of the current GCP instance"""
+
+    from psutil import virtual_memory
+    from psutil import cpu_count
+
+    def __init__(self):
+        self.__type = None
+        self.__cpu = None
+        self.__memoryGB = None
+        self.__nvmeDiskCount = None
+
+    @staticmethod
+    def is_gce_instance():
+      """Check if it's GCE instance via DNS lookup to metadata server."""
+      import socket
+      try:
+        addrlist = socket.getaddrinfo('metadata.google.internal', 80)
+      except socket.gaierror:
+        return False
+      for res in addrlist:
+        af, socktype, proto, canonname, sa = res
+        if (af == socket.AF_INET):
+          addr, port = sa
+          if (addr == "169.254.169.254"):
+            return True
+      return False
+
+    def __instance_metadata(self, path):
+        """query GCP metadata server, recursively!"""
+        return curl("http://169.254.169.254/computeMetadata/v1/instance/" + path+"?recursive=true", headers="Metadata-Flavor: Google")
+        #169.254.169.254 is metadata.google.internal
+
+    def _non_root_nvmes(self):
+        """get list of nvme disks from os, filter away if one of them is root"""
+        nvme_re = re.compile(r"nvme\d+n\d+$")
```

Comment:
Can we differentiate between local and remote disks?

Reply:
we are getting count of local nvmes from metadata and that is the source of truth - this code is mainly a check if nvme is used as root disk (fwiw, we do the same for AWS)
so source of truth for local nvme count are metadata (see below for the call to metadata parsing - __get_nvme_disks_from_metadata which is used in nvmeDiskCount, unless this list is smaller(which means they used local nvme for root disk))

Reply:
I didn't understand the bottom line, can we differentiate between local and remote disks?

Reply:
we can differentiate only based on seeing "nvme" since right now no other disk in GCP is creating nvme devices in /dev/
(and as pointed in other comments, I didn't find a way to do mapping between metadata about local disks and OS presented devices)

Reply:
Each time you use the word "right now" you can expect me to explain again that google can add new instance types and our integration has to work with them.

Reply:
fixed, I restricted the auto tuning code to only kick in for supported instances and anything else will trigger a new run of iotune